soh/src/code/z_parameter.c

```diff
@@ -1411,24 +1411,31 @@ void Inventory_SwapAgeEquipment(void) {
     u16 temp;
 
     if (LINK_AGE_IN_YEARS == YEARS_CHILD) {
-        // When becoming adult, remove swordless flag since we'll get master sword
-        // Only in rando to keep swordless link bugs in vanilla
-        if (gSaveContext.n64ddFlag) {
-            gSaveContext.infTable[29] &= ~1;
-        }
+        
 
         for (i = 0; i < ARRAY_COUNT(gSaveContext.equips.buttonItems); i++) {
             if (i != 0) {
                 gSaveContext.childEquips.buttonItems[i] = gSaveContext.equips.buttonItems[i];
             } else {
-                gSaveContext.childEquips.buttonItems[i] = ITEM_SWORD_KOKIRI;
+                if (CVarGetInteger("gSwitchAge", 0) && 
+                    gSaveContext.infTable[29]) {
```

Comment:
I know `infTable[29]` is strictly only used for the swordless flag, so this check works, but maybe it should be more explicit like `(gSaveContext.infTable[29] & 1)`  or `gSaveContext.infTable[29] != 0`

Reply:
alright yea, former makes the most sense for me if done for clarity